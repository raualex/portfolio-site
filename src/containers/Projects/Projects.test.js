import React from 'react';
import { Projects, mapStateToProps, mapDispatchToProps } from './';
import { shallow } from 'enzyme';

describe('Projects', () => {
  let wrapper;
  let mockFunc = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Projects setSelectedProj={mockFunc} selectedNav={'Projects'} />)
  });

  it('should match the snapshot if no project is selected', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot if a project is selected', () => {
    wrapper = shallow(<Projects selectedProj={'Bio'} selectedNav={'Projects'}  />)
    expect(wrapper).toMatchSnapshot()
  });

  describe('setProject function', () => {
    it('should fire setSelectedProj with the correct params', () => {
      wrapper.instance().setProject('Mentor Match')
      expect(mockFunc).toHaveBeenCalledWith('Mentor Match')
    });
  });
});

describe('mapStateToProps function', () => {
  it('should return an object with the selected project', () => {
    const mockState = {
      selectedProj: 'Mentor Match',
      somethingElse: 'w00t!'
    }
    const expected = {
      selectedProj: 'Mentor Match'
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps function', () => {
  const mockDispatch = jest.fn()
  let mappedProps;

  beforeEach(() => {
    mappedProps = mapDispatchToProps(mockDispatch)
  });

  it('should call dispatch when setSelectedProj is called', () => {
    mappedProps.setSelectedProj()
    expect(mockDispatch).toHaveBeenCalled()
  });
});