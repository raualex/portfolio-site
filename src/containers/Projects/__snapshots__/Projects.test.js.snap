// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Projects />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="project-links-box"
      >
        <div
          className="projects-titles-container"
        >
          <h1
            className="projects-title"
          >
            Recent and On-going Projects
          </h1>
          <h4>
            (Click the screenshot to see more info)
          </h4>
        </div>
        <div
          className="mentormatch-container"
          onClick={[Function]}
        >
          <h2>
            Mentor Match
          </h2>
          <img
            alt="mentor match screenshot"
            className="mm-screenshot"
            src="mentor-match-screenshot.png"
          />
        </div>
        <div
          className="ebmdb-container"
          onClick={[Function]}
        >
          <h2>
            EBM Database (in progress)
          </h2>
          <img
            alt="EBM Database screenshot"
            className="ebmdb-screenshot"
            src="ebm-database.png"
          />
        </div>
        <div
          className="bandr-container"
          onClick={[Function]}
        >
          <h2>
            Bread and Roses
          </h2>
          <img
            alt="bread and roses screenshot"
            className="bar-screenshot"
            src="breadandroses-screenshot.png"
          />
        </div>
        <div
          className="swapibox-container"
          onClick={[Function]}
        >
          <h2>
            SWAPIbox
          </h2>
          <img
            alt="swapi box screenshot"
            className="swapi-screenshot"
            src="swapibox-screenshot.png"
          />
        </div>
        <div
          className="gametime-container"
          onClick={[Function]}
        >
          <h2>
            Game Time - Snake
          </h2>
          <img
            alt="game time screenshot"
            className="gametime-screenshot"
            src="gametime-screenshot.jpg"
          />
        </div>
      </div>,
      "className": "projects-container blur",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="projects-titles-container"
          >
            <h1
              className="projects-title"
            >
              Recent and On-going Projects
            </h1>
            <h4>
              (Click the screenshot to see more info)
            </h4>
          </div>,
          <div
            className="mentormatch-container"
            onClick={[Function]}
          >
            <h2>
              Mentor Match
            </h2>
            <img
              alt="mentor match screenshot"
              className="mm-screenshot"
              src="mentor-match-screenshot.png"
            />
          </div>,
          <div
            className="ebmdb-container"
            onClick={[Function]}
          >
            <h2>
              EBM Database (in progress)
            </h2>
            <img
              alt="EBM Database screenshot"
              className="ebmdb-screenshot"
              src="ebm-database.png"
            />
          </div>,
          <div
            className="bandr-container"
            onClick={[Function]}
          >
            <h2>
              Bread and Roses
            </h2>
            <img
              alt="bread and roses screenshot"
              className="bar-screenshot"
              src="breadandroses-screenshot.png"
            />
          </div>,
          <div
            className="swapibox-container"
            onClick={[Function]}
          >
            <h2>
              SWAPIbox
            </h2>
            <img
              alt="swapi box screenshot"
              className="swapi-screenshot"
              src="swapibox-screenshot.png"
            />
          </div>,
          <div
            className="gametime-container"
            onClick={[Function]}
          >
            <h2>
              Game Time - Snake
            </h2>
            <img
              alt="game time screenshot"
              className="gametime-screenshot"
              src="gametime-screenshot.jpg"
            />
          </div>,
        ],
        "className": "project-links-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="projects-title"
              >
                Recent and On-going Projects
              </h1>,
              <h4>
                (Click the screenshot to see more info)
              </h4>,
            ],
            "className": "projects-titles-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recent and On-going Projects",
                "className": "projects-title",
              },
              "ref": null,
              "rendered": "Recent and On-going Projects",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(Click the screenshot to see more info)",
              },
              "ref": null,
              "rendered": "(Click the screenshot to see more info)",
              "type": "h4",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Mentor Match
              </h2>,
              <img
                alt="mentor match screenshot"
                className="mm-screenshot"
                src="mentor-match-screenshot.png"
              />,
            ],
            "className": "mentormatch-container",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mentor Match",
              },
              "ref": null,
              "rendered": "Mentor Match",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "mentor match screenshot",
                "className": "mm-screenshot",
                "src": "mentor-match-screenshot.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                EBM Database (in progress)
              </h2>,
              <img
                alt="EBM Database screenshot"
                className="ebmdb-screenshot"
                src="ebm-database.png"
              />,
            ],
            "className": "ebmdb-container",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "EBM Database (in progress)",
              },
              "ref": null,
              "rendered": "EBM Database (in progress)",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "EBM Database screenshot",
                "className": "ebmdb-screenshot",
                "src": "ebm-database.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Bread and Roses
              </h2>,
              <img
                alt="bread and roses screenshot"
                className="bar-screenshot"
                src="breadandroses-screenshot.png"
              />,
            ],
            "className": "bandr-container",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Bread and Roses",
              },
              "ref": null,
              "rendered": "Bread and Roses",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "bread and roses screenshot",
                "className": "bar-screenshot",
                "src": "breadandroses-screenshot.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                SWAPIbox
              </h2>,
              <img
                alt="swapi box screenshot"
                className="swapi-screenshot"
                src="swapibox-screenshot.png"
              />,
            ],
            "className": "swapibox-container",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SWAPIbox",
              },
              "ref": null,
              "rendered": "SWAPIbox",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "swapi box screenshot",
                "className": "swapi-screenshot",
                "src": "swapibox-screenshot.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Game Time - Snake
              </h2>,
              <img
                alt="game time screenshot"
                className="gametime-screenshot"
                src="gametime-screenshot.jpg"
              />,
            ],
            "className": "gametime-container",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game Time - Snake",
              },
              "ref": null,
              "rendered": "Game Time - Snake",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "game time screenshot",
                "className": "gametime-screenshot",
                "src": "gametime-screenshot.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="project-links-box"
        >
          <div
            className="projects-titles-container"
          >
            <h1
              className="projects-title"
            >
              Recent and On-going Projects
            </h1>
            <h4>
              (Click the screenshot to see more info)
            </h4>
          </div>
          <div
            className="mentormatch-container"
            onClick={[Function]}
          >
            <h2>
              Mentor Match
            </h2>
            <img
              alt="mentor match screenshot"
              className="mm-screenshot"
              src="mentor-match-screenshot.png"
            />
          </div>
          <div
            className="ebmdb-container"
            onClick={[Function]}
          >
            <h2>
              EBM Database (in progress)
            </h2>
            <img
              alt="EBM Database screenshot"
              className="ebmdb-screenshot"
              src="ebm-database.png"
            />
          </div>
          <div
            className="bandr-container"
            onClick={[Function]}
          >
            <h2>
              Bread and Roses
            </h2>
            <img
              alt="bread and roses screenshot"
              className="bar-screenshot"
              src="breadandroses-screenshot.png"
            />
          </div>
          <div
            className="swapibox-container"
            onClick={[Function]}
          >
            <h2>
              SWAPIbox
            </h2>
            <img
              alt="swapi box screenshot"
              className="swapi-screenshot"
              src="swapibox-screenshot.png"
            />
          </div>
          <div
            className="gametime-container"
            onClick={[Function]}
          >
            <h2>
              Game Time - Snake
            </h2>
            <img
              alt="game time screenshot"
              className="gametime-screenshot"
              src="gametime-screenshot.jpg"
            />
          </div>
        </div>,
        "className": "projects-container blur",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="projects-titles-container"
            >
              <h1
                className="projects-title"
              >
                Recent and On-going Projects
              </h1>
              <h4>
                (Click the screenshot to see more info)
              </h4>
            </div>,
            <div
              className="mentormatch-container"
              onClick={[Function]}
            >
              <h2>
                Mentor Match
              </h2>
              <img
                alt="mentor match screenshot"
                className="mm-screenshot"
                src="mentor-match-screenshot.png"
              />
            </div>,
            <div
              className="ebmdb-container"
              onClick={[Function]}
            >
              <h2>
                EBM Database (in progress)
              </h2>
              <img
                alt="EBM Database screenshot"
                className="ebmdb-screenshot"
                src="ebm-database.png"
              />
            </div>,
            <div
              className="bandr-container"
              onClick={[Function]}
            >
              <h2>
                Bread and Roses
              </h2>
              <img
                alt="bread and roses screenshot"
                className="bar-screenshot"
                src="breadandroses-screenshot.png"
              />
            </div>,
            <div
              className="swapibox-container"
              onClick={[Function]}
            >
              <h2>
                SWAPIbox
              </h2>
              <img
                alt="swapi box screenshot"
                className="swapi-screenshot"
                src="swapibox-screenshot.png"
              />
            </div>,
            <div
              className="gametime-container"
              onClick={[Function]}
            >
              <h2>
                Game Time - Snake
              </h2>
              <img
                alt="game time screenshot"
                className="gametime-screenshot"
                src="gametime-screenshot.jpg"
              />
            </div>,
          ],
          "className": "project-links-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="projects-title"
                >
                  Recent and On-going Projects
                </h1>,
                <h4>
                  (Click the screenshot to see more info)
                </h4>,
              ],
              "className": "projects-titles-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Recent and On-going Projects",
                  "className": "projects-title",
                },
                "ref": null,
                "rendered": "Recent and On-going Projects",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "(Click the screenshot to see more info)",
                },
                "ref": null,
                "rendered": "(Click the screenshot to see more info)",
                "type": "h4",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Mentor Match
                </h2>,
                <img
                  alt="mentor match screenshot"
                  className="mm-screenshot"
                  src="mentor-match-screenshot.png"
                />,
              ],
              "className": "mentormatch-container",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mentor Match",
                },
                "ref": null,
                "rendered": "Mentor Match",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "mentor match screenshot",
                  "className": "mm-screenshot",
                  "src": "mentor-match-screenshot.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  EBM Database (in progress)
                </h2>,
                <img
                  alt="EBM Database screenshot"
                  className="ebmdb-screenshot"
                  src="ebm-database.png"
                />,
              ],
              "className": "ebmdb-container",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "EBM Database (in progress)",
                },
                "ref": null,
                "rendered": "EBM Database (in progress)",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "EBM Database screenshot",
                  "className": "ebmdb-screenshot",
                  "src": "ebm-database.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Bread and Roses
                </h2>,
                <img
                  alt="bread and roses screenshot"
                  className="bar-screenshot"
                  src="breadandroses-screenshot.png"
                />,
              ],
              "className": "bandr-container",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Bread and Roses",
                },
                "ref": null,
                "rendered": "Bread and Roses",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "bread and roses screenshot",
                  "className": "bar-screenshot",
                  "src": "breadandroses-screenshot.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  SWAPIbox
                </h2>,
                <img
                  alt="swapi box screenshot"
                  className="swapi-screenshot"
                  src="swapibox-screenshot.png"
                />,
              ],
              "className": "swapibox-container",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SWAPIbox",
                },
                "ref": null,
                "rendered": "SWAPIbox",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "swapi box screenshot",
                  "className": "swapi-screenshot",
                  "src": "swapibox-screenshot.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Game Time - Snake
                </h2>,
                <img
                  alt="game time screenshot"
                  className="gametime-screenshot"
                  src="gametime-screenshot.jpg"
                />,
              ],
              "className": "gametime-container",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Game Time - Snake",
                },
                "ref": null,
                "rendered": "Game Time - Snake",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "game time screenshot",
                  "className": "gametime-screenshot",
                  "src": "gametime-screenshot.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
